### PR TITLE
fix: overlay and nfs don't work well together

### PR DESCRIPTION
**Pull request by rchincha**
Add a check for remote-mounted filesystems like NFS.

https://github.com/project-stacker/stacker/issues/289

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
```
$ ./stacker --roots-dir ~/tmp check
Linux 5.19.0-46-generic #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 21 15:35:31 UTC 2
/home/rchincha/tmp NFS(6969)
error: remote-mounted filesystem NFS(6969) detected at /home/rchincha/tmp
error: exit status 1
```

**Review by hallyn — APPROVED** [review]

File: cmd/stacker/check.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ func doCheck(ctx *cli.Context) error {
 
 	log.Infof("%s %s", config.RootFSDir, fstype)
 
+	if fstype == "NFS(6969)" {
+		return errors.Errorf("remote-mounted filesystem %s detected at %s", fstype, config.RootFSDir)
+	}
```
Comment (smoser) [inline cmd/stacker/check.go:44]: I'd like to suggest an improvement to the error message:

    stacker roots dir (--roots-dir) path %s is not supported on NFS.

the string "NFS(6969)" has exaxtly zero relevant hits in my search, so there is no value in showing it to a user.  and specifying *which* config option it came from is important to fix the issue.

**Review by smoser — CHANGES_REQUESTED** [review]
just a message cleanup.
but other than that, +1.

File: cmd/stacker/check.go (hunk 1 above)
Comment (rchincha) [reply cmd/stacker/check.go:44]:
```
$ ./stacker --roots-dir ~/tmp check
Linux 5.19.0-46-generic #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 21 15:35:31 UTC 2
/home/rchincha/tmp NFS(6969)
error: roots dir (--roots-dir) path /home/rchincha/tmp is not supported on NFS.
error: exit status 1
```

Fixed.

**Review by hallyn — APPROVED** [review]

**Pull request merged**